bindings/cpp/result_entry.cpp

```diff
@@ -354,14 +454,81 @@ void iterator_result_container::append(const dnet_iterator_response *response)
 //* Sort container by (key, timestamp) tuple
 void iterator_result_container::sort()
 {
-	int err;
-
 	if (m_sorted == true)
 		return;
 
-	err = dnet_iterator_response_container_sort(m_fd, m_write_position);
-	if (err != 0)
-		throw_error(err, "sort failed");
+	if (m_write_position % sizeof(struct dnet_iterator_response) != 0)
+		throw_error(-EINVAL, "invalid container size");
+
+	int err;
+	std::vector<std::shared_ptr<iterator_result_chunk> > chunks;
+
+	uint64_t offset = 0;
+	while (offset < m_write_position) {
+		const auto num_remaining_items = (m_write_position - offset) / sizeof(struct dnet_iterator_response);
+		const size_t num_chunk_items = std::min(num_remaining_items, MAX_ITERATOR_RESULT_ITEMS_IN_CHUNK);
+
+		auto chunk = std::make_shared<iterator_result_chunk>(m_fd, offset, num_chunk_items);
+		err = chunk->sort();
+		if (err != 0)
+			throw_error(err, "chunk sort failed");
+
+		chunks.push_back(chunk);
+		offset += num_chunk_items * sizeof(struct dnet_iterator_response);
+	}
+
+	if (chunks.size() > 1) {
+		const size_t num_buffer_items = std::max<size_t>(1, MAX_ITERATOR_RESULT_ITEMS_IN_CHUNK / chunks.size());
```

Comment:
To avoid using `set_num_buffer_items` and reading first buffer which could be taken from `sort`.